packages/eas-cli/src/build/metadata.ts

```diff
@@ -148,3 +149,16 @@ function resolveIosEnterpriseProvisioning(
 ): IosEnterpriseProvisioning | undefined {
   return ctx.buildProfile.enterpriseProvisioning;
 }
+
+export async function getFormattedCommitMessageAsync(
+  vcsClient: Client
+): Promise<string | undefined> {
+  const maybeCommitMessage = (await vcsClient.getLastCommitMessageAsync())?.trim();
+  if (!maybeCommitMessage) {
+    return undefined;
+  }
+  const lines = maybeCommitMessage.split('\n');
```

Comment:
I would leave it as it is and maybe format that depending on the context before displaying it to the user.
e.g. places that can only show one line should only display the first line of the commit message even if it can fit more, but in places where there is enough space for multiline text we should preserve the original formatting.